aws_lambda_powertools/utilities/_data_masking/provider/base.py

```diff
@@ -1,13 +1,70 @@
+from __future__ import annotations
+
 import json
-from typing import Any
+from typing import Any, Iterable, Union
 
 from aws_lambda_powertools.utilities._data_masking.constants import DATA_MASKING_STRING
 
 
 class BaseProvider:
     """
-    When you try to create an instance of a subclass that does not implement the encrypt method,
-    you will get a NotImplementedError with a message that says the method is not implemented:
+    The BaseProvider class serves as an abstract base class for data masking providers.
+
+    Attributes
+    ----------
+    json_serializer : Callable
+        A callable function responsible for JSON serialization.
+    json_deserializer : Callable
+        A callable function responsible for JSON deserialization.
+
+    Methods
+    -------
+    default_json_serializer(data)
+        Default method for JSON serialization.
+    default_json_deserializer(data)
+        Default method for JSON deserialization.
+    encrypt(data)
+        Abstract method for encrypting data. Subclasses must implement this method.
+    decrypt(data)
+        Abstract method for decrypting data. Subclasses must implement this method.
+    mask(data)
+        Default method for masking data.
+
+    Examples
```

Comment:
>  Implementation: Docstrings (some don't match rest of the code base)

Should I get rid of the "Example" parts of the docstrings? I think it's redundant since it's in the docs.

Reply:
It serves a different purpose -- customers wanting to stay within the IDE. One example with encryption using fields+encryption context should suffice. Like other features, the `Example` should work as-is if copied/pasted.